### PR TITLE
Fix #7183: Use MutationObserver for JitsiSDK iframe's src attribute.

### LINKED ISSUE #7183: Use mutationobserver to fetch BraveTalk native iframe's src attribute

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes #7183 

## Submitter Checklist:

- [ ] *Unit Tests* are updated to cover new or changed functionality
- [ ] User-facing strings use `NSLocalizableString()`
- [ ] New or updated UI has been tested across:
  - [ ] Light & dark mode
  - [ ] Different size classes (iPhone, landscape, iPad)
  - [ ] Different dynamic type sizes

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Review by Brandon-T — COMMENTED** [review]
Hmm maybe we could just override the `HTMLIFrameElement.setAttribute`

And check for "src" attribute being set. We do this in playlist atm. But not sure if we should for jitsi.

EDIT:
I guess this is fine since it only observes the frame and the attribute filter is also "src"

**Review by Brandon-T — APPROVED** [review]

**Comment by iccub** [comment]
@Brandon-T any particular benefit in that approach over mutationobserver?

**Comment by Brandon-T** [comment]
> @Brandon-T any particular benefit in that approach over mutationobserver? 

I was thinking about the performance of observers vs. a function that was made to handle attribute changes. But then I realized the observer actually is filtering it by that attribute anyway, so it should be similar.

So no difference for your approach.

For playlist since we observe the whole page it'd be very expensive. So instead we override src setter only for video and audio elements which is much better.

So TLDR: you're good. No worries. No need to change anything :)

**Comment by iccub** [comment]
Closing this one for now
